flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptor.java

```diff
@@ -141,9 +146,16 @@ public TaskDeploymentDescriptor(
 			List<BlobKey> requiredJarFiles, List<URL> requiredClasspaths,
 			int targetSlotNumber) {
 
-		this(appId, jobID, vertexID, executionId, taskName, indexInSubtaskGroup, numberOfSubtasks, attemptNumber,
-				jobConfiguration, taskConfiguration, invokableClassName, producedPartitions,
-				inputGates, requiredJarFiles, requiredClasspaths, targetSlotNumber, null, -1);
+		this(appId, jobID, vertexID, executionId, new ExecutionConfig(), taskName, indexInSubtaskGroup,
```

Comment:
This constructor only exists for tests. The executionConfig in the TDD is the change of this pull request, so here the new ExecutionConfig is just to not break the already existing tests that were testing other parts of the functionality of the TDD. New tests are added to test the new functionality.